subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -163,6 +163,193 @@ class ArtifactTransformIntegrationTest extends AbstractIntegrationSpec {
         file("app/build/libs").listFiles().size() == 1
     }
 
+    def "User gets a reasonable error message when a transformation throws exception"() {
+        given:
+        executer.withStackTraceChecksDisabled()
+        buildFile << """
+            import org.gradle.api.artifacts.transform.*
+
+            apply plugin: 'java'
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                compile 'com.google.guava:guava:19.0'
+            }
+
+            configurations {
+                hash {
+                    extendsFrom(configurations.compile)
+                    format = 'md5'
+                    resolutionStrategy.registerTransform(TransformWithIllegalArgumentException) { }
+                }
+            }
+
+            task resolve(type: Copy) {
+                from configurations.hash.incoming.artifacts*.file
+                into "\${buildDir}/libs"
+            }
+
+            @TransformInput(format = 'jar')
+            class TransformWithIllegalArgumentException extends ArtifactTransform {
+
+                @TransformOutput(format = 'md5')
+                File getOutput() {
+                    return null
+                }
+
+                void transform(File input) {
+                    throw new IllegalArgumentException("Transform Implementation Missing!")
+                }
+            }
+        """
+
+        when:
+        succeeds "resolve"
+
+        then:
+        errorOutput.contains("Error while transforming 'guava-19.0.jar' to format 'md5' using 'TransformWithIllegalArgumentException' - Transform Implementation Missing!")
```

Comment:
should use the `ExecutionFailure` test fixture (via the `failure` property on the test class) to check that the exception was reported correctly, rather than scraping the output.